### PR TITLE
TCK part of #209: "LRA_HTTP_RECOVERY_HEADER is mandatory"

### PR DESCRIPTION
**Pull request by rdebusscher**
#209

### CODE REVIEW

**Comment by rdebusscher** [comment]
@xstefank also for you to review ...

File: tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java [hunk 1]
```diff
@@ -96,7 +96,12 @@
     @Produces(MediaType.APPLICATION_JSON)
     @Status
     @LRA(value = LRA.Type.NOT_SUPPORTED)
-    public Response status(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
+    public Response status(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId,
+                           @HeaderParam(LRA_HTTP_RECOVERY_HEADER) URI recoveryId) {
+
+        assertHeaderPresent(lraId, LRA_HTTP_CONTEXT_HEADER); // the TCK expects the coordinator to invoke @Completed methods
```
Comment (xstefank) [inline tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java]: The comment should not mention coordinator and this is Status method but the comment is mentioning Complete. I guess probably a copy-paste error?

File: tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java [hunk 2]
```diff
@@ -135,10 +140,13 @@ public Response leaveWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
...
     @Complete
-    public Response completeWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId, String userData) {
+    public Response completeWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId,
+                                 @HeaderParam(LRA_HTTP_RECOVERY_HEADER) URI recoveryId,
+                                 String userData) {
         lraMetricService.incrementMetric(LRAMetricType.Completed, lraId, LraResource.class.getName());
 
-        assertHeaderPresent(lraId); // the TCK expects the coordinator to invoke @Completed methods
+        assertHeaderPresent(lraId, LRA_HTTP_CONTEXT_HEADER); // the TCK expects the coordinator to invoke @Completed methods
+        assertHeaderPresent(recoveryId, LRA_HTTP_RECOVERY_HEADER); // the TCK expects the coordinator to invoke @Completed methods
```
Comment (xstefank) [inline tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java]: same as above, we should not mention coordinator explicitly, and typo "`@Complete` methods"

File: tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java [hunk 3]
```diff
@@ -163,9 +171,12 @@ public Response completeWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId, St
...
     @Produces(MediaType.APPLICATION_JSON)
     @Compensate(timeLimit = 0, timeUnit = ChronoUnit.SECONDS)  // O and ChronoUnit.SECONDS are the default, just to be explicit here.
-    public Response compensateWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId, String userData) {
+    public Response compensateWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId,
+                                   @HeaderParam(LRA_HTTP_RECOVERY_HEADER) URI recoveryId,
+                                   String userData) {
 
-        assertHeaderPresent(lraId); // the TCK expects the coordinator to invoke @Compensated methods
+        assertHeaderPresent(lraId, LRA_HTTP_CONTEXT_HEADER); // the TCK expects the coordinator to invoke @Compensated methods
+        assertHeaderPresent(recoveryId, LRA_HTTP_RECOVERY_HEADER); // the TCK expects the coordinator to invoke @Compensated methods
```
Comment (xstefank) [inline tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java]: `@Compesate` methods

File: tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java (hunk 1 above)
Comment (rdebusscher) [reply tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java]: I just copied the comment from another `assertHeaderPresent` where the term coordinator was already used. 
But indeed, we should no longer use the term.
I'll update (also the copy-paste errors!)
Thx

**Comment by mmusgrov** [comment]
@xstefank Rudy has addressed your review comments

**Comment by xstefank** [comment]
@mmusgrov approved, Thanks @rdebusscher !

**Pull request merged**